File: ios/Sources/VoiceRecorder/Core/CurrentRecordingStatus.swift
import Foundation

/// Represents the current recording state.
enum CurrentRecordingStatus: String {

    case RECORDING
    case PAUSED
    case INTERRUPTED
    /// The session died while it was running (see `RecordingFailure`). A state
    /// of its own so that "not NONE" keeps meaning "there is a file to collect".
    case ERROR
    case NONE

}

File: android/src/main/java/app/independo/capacitorvoicerecorder/core/CurrentRecordingStatus.java
package app.independo.capacitorvoicerecorder.core;

/** Represents the current recording state. */
public enum CurrentRecordingStatus {
    RECORDING,
    PAUSED,
    INTERRUPTED,
    /**
     * The session died while it was running (see {@link RecordingFailure}).
     *
     * A state of its own rather than a flag beside `RECORDING`, because callers
     * branch on this: "not NONE" has to keep meaning "there is a file to collect",
     * so that stopping a failed session still hands back whatever was captured.
     */
    ERROR,
    NONE
}
